code/modules/industrial_lift/crossing_signal.dm

```diff
@@ -103,37 +103,43 @@
 	end_processing()
 
 /obj/machinery/crossing_signal/process()
-	var/obj/structure/industrial_lift/tram/central/tram_part = tram_ref?.resolve()
+	var/datum/lift_master/tram/tram = tram_ref?.resolve()
 
 	// Check for stopped states.
-	if(!tram_part || !is_operational)
+	if(!tram || !is_operational)
 		// Tram missing, or we lost power.
 		// Tram missing is always safe (green)
 		set_signal_state(XING_STATE_GREEN, force = !is_operational)
 		return PROCESS_KILL
 
 	use_power(active_power_usage)
 
+	var/obj/structure/industrial_lift/tram/tram_part = tram.return_closest_platform_to(src)
+
+	if(QDELETED(tram_part))
+		set_signal_state(XING_STATE_GREEN, force = !is_operational)
```

Comment:
We should probably convert this to signals and de-hardcode this, that way more things can listen for tram status.

Reply:
im not sure what you mean by this. what part should be converted to signals? the tram proximity check? if so sure, not in this pr but you could register to the tram entering turfs within a certain proximity and use that to determine the current state. not sure what you mean by de-hardcode though